cpp_modules/src/benchmarks/mcts_benchmark.cpp

```diff
@@ -73,53 +135,334 @@ void dumbBenchmark() {
-	std::vector<int> results;
 
-	int result;
+	std::vector<float> results;
+
+	float result = 0;
+	int benchmarkIndex = 0;
+
+	benchmarkIndex++;
+	result = benchmarkOnTime({
+		.benchmarkIndex = benchmarkIndex,
+		.seed = seed,
+		.numberOfRollouts = 500,
+		.endTime = 300,
+		.exploration = sqrt(2),
+		.valueHeuristic = ValueHeuristic::UCT,
+		.rolloutHeuristic = RolloutHeuristic::Random,
 
-	// Benchmark 1: UCT with Random as the rollout heuristic
-	/*result = benchmark(1, seed, numberOfActions, numberOfRollouts, rolloutDepth, exploration, valueHeuristic,
-	                   rolloutHeuristic);
+	});
 	results.push_back(result);
 
-	// Benchmark 2: UCT with weighted choice as the rollout heuristic
-	rolloutHeuristic = RolloutHeuristic::WeightedChoice;
+	benchmarkIndex++;
+	result = benchmarkOnTime({
+		.benchmarkIndex = benchmarkIndex,
+		.seed = seed,
+		.numberOfRollouts = 1000,
+		.endTime = 300,
+		.exploration = sqrt(2),
+		.valueHeuristic = ValueHeuristic::UCT,
+		.rolloutHeuristic = RolloutHeuristic::Random,
+	});
+	results.push_back(result);
 
-	result = benchmark(2, seed, numberOfActions, numberOfRollouts, rolloutDepth, exploration, valueHeuristic,
-	                   rolloutHeuristic, true);
-	results.push_back(result);*/
+	benchmarkIndex++;
+	result = benchmarkOnTime({
+		.benchmarkIndex = benchmarkIndex,
+		.seed = seed,
+		.numberOfRollouts = 2000,
+		.endTime = 300,
+		.exploration = sqrt(2),
+		.valueHeuristic = ValueHeuristic::UCT,
+		.rolloutHeuristic = RolloutHeuristic::Random,
+		.shouldPrintActions = false
+	});
+	results.push_back(result);
 
-	// Benchmark 3: UCB1-Normal2 with random as the rollout heuristic
-	rolloutHeuristic = RolloutHeuristic::Random;
-	result = benchmark(3, seed, numberOfActions, numberOfRollouts, rolloutDepth, exploration, valueHeuristic,
-	                   rolloutHeuristic);
+	benchmarkIndex++;
+	result = benchmarkOnTime({
+		.benchmarkIndex = benchmarkIndex,
+		.seed = seed,
+		.numberOfRollouts = 500,
+		.endTime = 300,
+		.exploration = sqrt(2),
+		.valueHeuristic = ValueHeuristic::UCT,
+		.rolloutHeuristic = RolloutHeuristic::WeightedChoice,
+	});
 	results.push_back(result);
 
-	valueHeuristic = ValueHeuristic::Ucb1Normal2;
-	result = benchmark(3, seed, numberOfActions, numberOfRollouts, rolloutDepth, exploration, valueHeuristic,
-	                   rolloutHeuristic);
+	benchmarkIndex++;
+	result = benchmarkOnTime({
+		.benchmarkIndex = benchmarkIndex,
+		.seed = seed,
+		.numberOfRollouts = 1000,
+		.endTime = 300,
+		.exploration = sqrt(2),
+		.valueHeuristic = ValueHeuristic::UCT,
+		.rolloutHeuristic = RolloutHeuristic::WeightedChoice,
+	});
 	results.push_back(result);
 
+	benchmarkIndex++;
+	result = benchmarkOnTime({
+		.benchmarkIndex = benchmarkIndex,
+		.seed = seed,
+		.numberOfRollouts = 2000,
+		.endTime = 300,
+		.exploration = sqrt(2),
+		.valueHeuristic = ValueHeuristic::UCT,
+		.rolloutHeuristic = RolloutHeuristic::WeightedChoice,
+	});
+	results.push_back(result);
+	//
```

Comment:
Is this being saved because we wanna use it again, or is it an oopsie?

Reply:
we might use it again, but i dont want it to run every time